server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1971,21 +1972,26 @@ public Response bind(
         try {
             consumerBindUtil.validateServiceLevel(consumer.getOwner(), serviceLevel);
             dryRunPools = entitler.getDryRun(consumer, serviceLevel);
-            if (dryRunPools == null) {
-                return new ArrayList<PoolQuantity>();
-            }
         }
         catch (ForbiddenException fe) {
-            return dryRunPools;
+            return Collections.<PoolQuantityDTO>emptyList();
         }
         catch (BadRequestException bre) {
             throw bre;
         }
         catch (RuntimeException re) {
-            return dryRunPools;
+            return Collections.<PoolQuantityDTO>emptyList();
+        }
+        if (dryRunPools != null) {
+            List<PoolQuantityDTO> dryRunPoolDtos = new ArrayList<PoolQuantityDTO>();
+            for (PoolQuantity pq : dryRunPools) {
+                dryRunPoolDtos.add(this.translator.translate(pq, PoolQuantityDTO.class));
+            }
+            return dryRunPoolDtos;
```

Comment:
This is what I'd do here as well, but eventually we should replace this block with either the streaming stuff, a translation iterator or whatever we come up with for bulk translation. For now maybe a TODO to improve this in the future.